app/scripts/metamask-controller.js

```diff
@@ -643,6 +643,31 @@ export default class MetamaskController extends EventEmitter {
         network.rpcEndpoints[network.defaultRpcEndpointIndex].networkClientId;
     }
 
+    // Fix the network controller state (selectedNetworkClientId) if it is invalid and report the error
+    if (
+      initialNetworkControllerState.networkConfigurationsByChainId &&
```

Comment:
The previous block of code forces `initialNetworkControllerState` to be defined.